src/Raven.Server/Documents/Indexes/Static/Counters/CounterItemFilterBehavior.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Indexes.Static.Counters
+{
+    public class CounterItemFilterBehavior : IIndexItemFilterBehavior
+    {
+        private readonly HashSet<LazyStringValue> _seenIds = new HashSet<LazyStringValue>();
```

Comment:
TODO:
- this is just an optimization, maybe we do not need to keep everything, just last X items
- check if we are not disposing LSV